### PR TITLE
 Fixes all remaining found issues with last nickname commit, Also impliements finalised version of PR 1163 

### PR DESCRIPTION
**Pull request by hitime1234**
Changes:
removes and adds fixes to rest of issues found see:
https://github.com/UniversityRadioYork/MyRadio/pull/1159
https://github.com/UniversityRadioYork/MyRadio/pull/1154
for more detail

Other changes impliemented a finalised version of
https://github.com/UniversityRadioYork/MyRadio/pull/1153

365391425-1be26c26-fd56-4c8d-b24e-c7626f143906

Let me know if I missed any

Squashed and move to new branch to so history doesn't get wack :3

### CODE REVIEW

File: docs/install.md [hunk 1]
```diff
@@ -29,8 +29,7 @@ If you have Docker on your system, use Docker Compose to set up an environment.
 Simply run `docker compose up -d` and visit "https://localhost:4443/myradio/".
 
 If you encounter an error with autoload.php:
-find the id of your myradio container using `docker dontainer ls`
-enter a bash session with `docker exec -it [myradioid] bash`
+enter a bash session with `docker compose exec myradio bash`
```
Comment (ashhhleyyy) [inline docs/install.md:32]: more concise to do `docker compose exec myradio composer install`

File: src/Public/js/myradio.scheduler.termlist.js [hunk 2]
```diff
@@ -1,26 +1,26 @@
...
+      "bVisible": false
+    },
+    //start
+    {
+      "sTitle": "Start Date"
     }
   ],
   "bPaginate": true
-});
+});
```
Comment (ashhhleyyy) [inline src/Public/js/myradio.scheduler.termlist.js:26]: this should be a separate PR

File: src/Public/js/myradio.timeslot.js [hunk 3]
```diff
@@ -1,3 +1,14 @@
+//Name helper function for concating names;
+function NameHelper(data) {
```
Comment (ashhhleyyy) [inline src/Public/js/myradio.timeslot.js:2]: maybe a better name for this (`formatName`? or something along those lines)

File: src/Classes/ServiceAPI/MyRadio_User.php [hunk 4]
```diff
@@ -442,7 +442,7 @@ public function getFName()
     }
 
      /**
-     * Returns the User's first name.
+     * Returns the User's nickname.
      *
      * @return string The User's first name
```
Comment (ashhhleyyy) [inline src/Classes/ServiceAPI/MyRadio_User.php:447]: this still needs changing

**Review by ashhhleyyy — CHANGES_REQUESTED** [review]

File: docs/install.md [hunk 5]
```diff
@@ -176,42 +175,7 @@ To do this, you first need to:
 -   Apply for a Season of your new Show (List My Shows -> New Season)
 -   Schedule the Season (Shows Scheduler)
 
-### Setting up your own 2016-site
```
Comment (ashhhleyyy) [inline docs/install.md:179]: is there a PR to add these to the 2016-site repo?

**Review by ashhhleyyy — COMMENTED** [review]
you've also missed
https://github.com/UniversityRadioYork/MyRadio/pull/1154/files#r1731837216

File: docs/install.md (hunk 5 above)
Comment (hitime1234) [reply docs/install.md:179]: PR has been added 
https://github.com/UniversityRadioYork/2016-site/pull/372
Also is this what you meant by the comment above or did you mean something else I missed

File: src/Classes/ServiceAPI/MyRadio_User.php (hunk 4 above)
Comment (hitime1234) [reply src/Classes/ServiceAPI/MyRadio_User.php]: whoops fixed it

File: src/Public/js/myradio.timeslot.js (hunk 3 above)
Comment (hitime1234) [reply src/Public/js/myradio.timeslot.js]: Change name to formatName as that works

File: src/Public/js/myradio.scheduler.termlist.js (hunk 2 above)
Comment (hitime1234) [reply src/Public/js/myradio.scheduler.termlist.js]: removed will add to seperate pr

File: docs/install.md (hunk 1 above)
Comment (hitime1234) [reply docs/install.md]: should be changed

File: docs/install.md (hunk 5 above)
Comment (ashhhleyyy) [reply docs/install.md:179]: nope, the comment is about something different (i linked to a specific comment)

File: src/Public/js/myradio.timeslot.js [hunk 6]
```diff
@@ -1,3 +1,14 @@
+//Format names function for concating names;
```
Comment (ashhhleyyy) [inline src/Public/js/myradio.timeslot.js:1]: not sure the comment is really necessary, and it reads quite confusingly to me

Comment (hitime1234) [reply src/Public/js/myradio.timeslot.js:1]: `//combines the first,nickname,last name of the user together`
would this be good or show I remove the comment fully

Comment (ashhhleyyy) [reply src/Public/js/myradio.timeslot.js]: probably best to just yeet it

File: src/Classes/ServiceAPI/MyRadio_User.php [hunk 7]
```diff
@@ -472,10 +472,8 @@ public function getName()
     {
         if (empty($this->nname)) {
             return $this->fname.' '.$this->sname;
-        } else {
-            return $this->fname.' "'.$this->nname.'" '.$this->sname;
-        }
-        return $this->fname.' '.$this->nname.' '.$this->sname;
+        } 
+        return $this->fname.' "'.$this->nname.'" '.$this->sname;
```
Comment (ashhhleyyy) [inline src/Classes/ServiceAPI/MyRadio_User.php:476]: it feels like this case should be inverted, as having a nickname set is definitely the non-default case
